packages/office-ui-fabric-react/src/components/List/List.tsx

```diff
@@ -1,1046 +1,44 @@
-      this._measureVersion++;
-    }
-
-    this._scrollHeight = scrollHeight;
-
-    // If the surface is above the container top or below the container bottom, or if this is not the first
-    // render return empty rect.
-    // The first time the list gets rendered we need to calculate the rectangle. The width of the list is
-    // used to calculate the width of the list items.
-    const visibleTop = Math.max(0, -surfaceRect.top);
-    const visibleRect = {
-      top: visibleTop,
-      left: surfaceRect.left,
-      bottom: visibleTop + window.innerHeight,
-      right: surfaceRect.right,
-      width: surfaceRect.width,
-      height: window.innerHeight
-    };
-
-    // The required/allowed rects are adjusted versions of the visible rect.
-    this._requiredRect = _expandRect(visibleRect, this._requiredWindowsBehind, this._requiredWindowsAhead);
-    this._allowedRect = _expandRect(visibleRect, renderedWindowsBehind!, renderedWindowsAhead!);
+    return <div>{pages}</div>;
   }
 }
 
-function _expandRect(rect: IRectangle, pagesBefore: number, pagesAfter: number): IRectangle {
-  const top = rect.top - pagesBefore * rect.height;
-  const height = rect.height + (pagesBefore + pagesAfter) * rect.height;
-
-  return {
-    top: top,
-    bottom: top + height,
-    height: height,
-    left: rect.left,
-    right: rect.right,
-    width: rect.width
-  };
-}
-
-function _isContainedWithin(innerRect: IRectangle, outerRect: IRectangle): boolean {
-  return (
-    innerRect.top >= outerRect.top &&
-    innerRect.left >= outerRect.left &&
-    innerRect.bottom! <= outerRect.bottom! &&
-    innerRect.right! <= outerRect.right!
-  );
-}
-
-function _mergeRect(targetRect: IRectangle, newRect: IRectangle): IRectangle {
-  targetRect.top = newRect.top < targetRect.top || targetRect.top === -1 ? newRect.top : targetRect.top;
-  targetRect.left = newRect.left < targetRect.left || targetRect.left === -1 ? newRect.left : targetRect.left;
-  targetRect.bottom =
-    newRect.bottom! > targetRect.bottom! || targetRect.bottom === -1 ? newRect.bottom : targetRect.bottom;
-  targetRect.right = newRect.right! > targetRect.right! || targetRect.right === -1 ? newRect.right : targetRect.right;
-  targetRect.width = targetRect.right! - targetRect.left + 1;
-  targetRect.height = targetRect.bottom! - targetRect.top + 1;
+function _derivePages(props: IListProps): JSX.Element[] {
+  const { items = [], onRenderCell } = props;
+  const itemsPerPage = 20;
+  const pages: JSX.Element[] = [];
+
+  for (let startIndex = 0; startIndex < items.length; startIndex += itemsPerPage) {
+    pages.push(
+      <ListPage
+        key={String(pages.length)}
+        items={items}
+        visible={true}
+        startIndex={startIndex}
+        endIndex={Math.min(startIndex + itemsPerPage - 1, items.length - 1)}
+        onRenderCell={onRenderCell}
+      />
+    );
+  }
 
-  return targetRect;
+  return pages;
```

Comment:
Did you actually mean to check this in? What is List2.tsx? (I can't actually leave a comment on that)

Reply:
I will check all the List changes. a PR got mixed together with an experiment.  Thanks for this!!